public/app/features/scenes/core/SceneDataProviderNode.tsx

```diff
@@ -0,0 +1,155 @@
+import { cloneDeep } from 'lodash';
+import React from 'react';
+// import { PanelData } from '@grafana/data';
+import { Unsubscribable } from 'rxjs/internal/types';
+
+import { DataQueryExtended, SceneQueryRunner } from '../querying/SceneQueryRunner';
+
+import { SceneDataObject, SceneObjectBase } from './SceneObjectBase';
+import { SceneTimeRange } from './SceneTimeRange';
+import { SceneDataState, SceneLayoutState, SceneObject, SceneObjectStatePlain } from './types';
+
+type SceneDataProviderInputParams = {
+  timeRange: SceneTimeRange;
+} & {
+  [key: string]: SceneObject;
+};
```

Comment:
Exploring a way of providing object's dependencies. Have not thought yet about how those dependencies would be serialized/deserialized later on.